src/ensembl/src/content/app/browser/Browser.tsx

```diff
@@ -127,20 +132,29 @@ export const Browser: FunctionComponent<BrowserProps> = (
       chrLocation
     };
 
-    props.setDataFromUrlAndSave(payload);
-
-    if (chrLocation) {
-      dispatchBrowserLocation(genomeId, chrLocation);
-    } else if (focus) {
+    if (focus && !chrLocation) {
+      /*
+       changeFocusObject needs to be called before setDataFromUrlAndSave
+       in order to prevent creating an previouslyViewedObject entry
+       for the focus object that is viewed first.
+       */
       props.changeFocusObject(focus);
+    } else if (focus && chrLocation) {
+      props.changeFocusObject(focus);
+      dispatchBrowserLocation(genomeId, focus, chrLocation);
+    } else if (chrLocation) {
+      dispatchBrowserLocation(genomeId, focus, chrLocation);
     }
+
+    props.setDataFromUrlAndSave(payload);
   };
 
   const dispatchBrowserLocation = (
     genomeId: string,
+    focus: string | null,
     chrLocation: ChrLocation
   ) => {
-    props.changeBrowserLocation(genomeId, chrLocation);
+    props.changeBrowserLocation(genomeId, focus, chrLocation);
```

Comment:
Can we change the function signature for changeBrowserLocation and dispatchBrowserLocation, so that instead of a list of arguments they accept an object with an optional field?

calling a function like this:

`dispatchBrowserLocation({genomeId: 'blah', chrLocation: 'something'})` is much more readable than `dispatchBrowserLocation(genomeId, null, chrLocation)` because it avoids asking the question "what on earth does the `null`" do?

Reply:
That's done